### PR TITLE
Standard Implementation of Custom Global Filter Handler

### PR DESCRIPTION
**Pull request by theycantrevealus**
Please review this implementation for global error handling usage. For now, it's applied for account module. Since we need to categorize and define some error code, I think we need to set a handler to centralize this requirement.

The problem is, the error code is already defined in service with specific condition
image

Current error class is `Error` which can only pass string parameter. We need to create custom class that can pass several parameter.

### CODE REVIEW

File: server/apps/core/src/main.ts [hunk 1]
```diff
@@ -67,13 +50,8 @@ async function bootstrap() {
...
   } else {
     fastifyAdapter = new FastifyAdapter({
-      logger: {
-        customLevels: {
-          log: 1,
-        },
-        file: `logs/gateway_${now.getFullYear()}_${now.getMonth()}_${now.getDate()}.log`,
-      },
-      // disableRequestLogging: true,
+      logger: false,
```
Comment (theycantrevealus) [inline server/apps/core/src/main.ts:53]: We need to disable logger here to  avoid fastify logger (Fastify using pinia and we prefer to use winston)

File: server/apps/core/src/main.ts [hunk 2]
```diff
@@ -83,14 +61,17 @@ async function bootstrap() {
...
-      // bodyParser: false,
-      logger: ['verbose', 'error'],
+      logger: ['verbose', 'error', 'warn'],
     }
   )
 
   const configService = app.get<ConfigService>(ConfigService)
 
-  const logger = app.get(WINSTON_MODULE_NEST_PROVIDER)
+  const logger = winston.createLogger({
```
Comment (theycantrevealus) [inline server/apps/core/src/main.ts:70]: Standalone logger is added due to filter class need

File: server/apps/core/src/main.ts [hunk 3]
```diff
@@ -10,43 +10,26 @@ import {
...
-          // },
-        },
-        customLevels: {
-          log: 1,
-        },
-        file: `logs/gateway_${now.getFullYear()}_${now.getMonth()}_${now.getDate()}.log`,
-      },
-      // disableRequestLogging: true,
+      logger: false,
+      disableRequestLogging: true,
```
Comment (theycantrevealus) [inline server/apps/core/src/main.ts:32]: Disable fastify request logging on gateway controller

File: server/apps/core/src/modules/account/dto/account.add.ts [hunk 4]
```diff
@@ -76,16 +70,18 @@ export class AccountAddDTO {
...
+  access: CMenu[]
 
   @ApiProperty({
-    type: [MenuPermissionJoin],
+    type: CMenuPermission,
+    isArray: true,
   })
   @IsNotEmpty()
-  permission: IMenuPermission[]
+  permission: CMenuPermission[]
```
Comment (theycantrevealus) [inline server/apps/core/src/modules/account/dto/account.add.ts:84]: We can't use mongoose raw object here

File: server/apps/core/src/modules/account/test/account.controller.e2e-spec.ts [hunk 5]
```diff
@@ -83,11 +95,15 @@ describe('Account Controller', () => {
...
+          headers: {
+            authorization: 'Bearer ey...',
+          },
           url: '/account',
           query: `lazyEvent=${ApiQueryGeneral.primeDT.example}`,
         })
         .then((result) => {
-          expect(result.statusCode).toEqual(200)
+          expect(result.statusCode).toEqual(HttpStatus.OK)
+          expect(logger.verbose).toHaveBeenCalled()
```
Comment (theycantrevealus) [inline server/apps/core/src/modules/account/test/account.controller.e2e-spec.ts:106]: Test if logger has been called when endpoint is hit

File: server/apps/filters/error.ts [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+        body: request.body ?? {},
+        params: request.params,
+        query: request.query,
+      },
+      result: responseSet,
+      account: account,
+      time: TM.getTimezone('Asia/Jakarta'),
+    }
+
+    console.log(dataSet)
```
Comment (theycantrevealus) [inline server/apps/filters/error.ts:55]: Prefer to remove

File: server/apps/utility/logger/interfaces.ts [hunk 7]
```diff
@@ -25,3 +26,15 @@ export interface WinstonModuleAsyncOptions
   inject?: any[]
   useClass?: Type<WinstonModuleOptionsFactory>
 }
+
+export interface HorasLogging {
```
Comment (theycantrevealus) [inline server/apps/utility/logger/interfaces.ts:30]: Is it a good name?

**Review by theycantrevealus — COMMENTED** [review]
Might need to enhance some changes

File: server/apps/core/src/modules/account/dto/account.add.ts (hunk 4 above)
Comment (tukangk3tik) [reply server/apps/core/src/modules/account/dto/account.add.ts:84]: Of course. Must create new object specific for dto.

**Comment by tukangk3tik** [comment]
> Please review this implementation for global error handling usage. For now, it's applied for account module. Since we need to categorize and define some error code, I think we need to set a handler to centralize this requirement.
> 
> The problem is, the error code is already defined in service with specific condition image
> 
> Current error class is `Error` which can only pass string parameter. We need to create custom class that can pass several parameter.

Why we don't create and throw CustomException (implement HttpException) to return error?

**Comment by theycantrevealus** [comment]
> > Please review this implementation for global error handling usage. For now, it's applied for account module. Since we need to categorize and define some error code, I think we need to set a handler to centralize this requirement.
> > The problem is, the error code is already defined in service with specific condition image
> > Current error class is `Error` which can only pass string parameter. We need to create custom class that can pass several parameter.
> 
> Why we don't create and throw CustomException (implement HttpException) to return error?

We do. But still not defined the code on filter class

**Pull request merged**